app/src/main/java/com/example/topmovies/fragment/MoviesFragment.kt

```diff
@@ -0,0 +1,82 @@
+package com.example.topmovies.fragment
+
+import android.content.SharedPreferences
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.os.bundleOf
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.topmovies.R
+import com.example.topmovies.adapter.MoviesAdapter
+import com.example.topmovies.databinding.FragmentMoviesBinding
+import org.koin.android.ext.android.inject
+import org.koin.androidx.viewmodel.ext.android.sharedViewModel
+
+class MoviesFragment : BaseFragment() {
+    private lateinit var binding: FragmentMoviesBinding
+    private val moviesAdapter by lazy { MoviesAdapter { id -> onClickItem(id) } }
+    private val moviesViewModel by sharedViewModel<MovieViewModel>()
+    private val sharedPref: SharedPreferences by inject()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        binding = FragmentMoviesBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        hideUpButton()
+        setupUI()
+        setupViewModel()
+    }
+
+    override fun onDestroyView() {
+        remove()
+        moviesViewModel.favoriteMovies.value?.forEach { movie -> saveFavoriteMovie(movie.id) }
+        super.onDestroyView()
+    }
+
+    private fun onClickItem(movieId: String) {
+        startMovieDetailsFragment(movieId)
+    }
+
+    private fun setupViewModel() {
+        moviesViewModel.apply {
+            if (moviesViewModel.movies.value == null) getMovies(getFavoriteMoviesId())
```

Comment:
Why would you do that?
Is there any other logical way to place to call the `getMovies()`?